utils/rjson.cc

```diff
@@ -193,6 +193,12 @@ struct guarded_yieldable_json_handler : public Handler {
 };
 
 void* internal::throwing_allocator::Malloc(size_t size) {
+    // For bypassing the address sanitizer failure in debug mode
+    #ifdef DEBUG
```

Comment:
I think DEBUG is the wrong macro to check - we also have a "SANITIZE" macro. I think at least theoretically, someone might build Scylla with sanitizer enabled but not other parts of "debug mode". I see we haven't been using the "SANITIZE" macro a lot, but maybe we should.